### ISSUE TITLE
read_metadata: Inconsistent behavior with missing and extra metadata columns

### ISSUE THREAD

**Issue by victorlin** [nextstrain/augur#1195]
### Current Behavior

When `augur.io.read_metadata()` is used on a file that has a row with a missing value, there are no warnings/errors, and columns are populated in the order of the header line.

When used on a file that has a row with an extra value, a cryptic and unhandled `KeyError` is raised.

### Expected behavior

The `KeyError` should be handled properly and with a more meaningful error message. I'd also think that if extra values result in errors, missing values should do the same since it might not be clear which column is missing.

### How to reproduce

See this [unmerged Cram test](https://github.com/nextstrain/augur/blob/731d9c532f20a9f458fc5ac20d4eab2b4a3ad9ca/tests/functional/filter/cram/filter-metadata-column-mismatch.t).

### Additional information

At first, I thought the `KeyError` behavior was related to [`pandas.read_csv()`](https://pandas.pydata.org/pandas-docs/version/1.5/reference/api/pandas.read_csv.html)'s `on_bad_lines='error'` default. However, the `KeyError` actually comes from setting `index_col` here:

https://github.com/nextstrain/augur/blob/73aad80fcb371c36ed93566803e8aaaf7ab53378/augur/io/metadata.py#L101-L102

**Comment by tsibley** [comment]
When you say "row with a missing value", do you mean a header row of N columns and a data row of <N columns?

**Comment by victorlin** [comment]
> When you say "row with a missing value", do you mean a header row of N columns and a data row of <N columns?

@tsibley yes, and same thing for extra value, meaning data row of >N columns.
